File: 240428/국영수 순이지/korean-english-math-order.py
n = int(input())

class student:
    def __init__(self, name, kor, eng, math):
        self.name = name
        self.kor = kor
        self.eng = eng
        self.math = math

students = []

for i in range(n):
    name, kor, eng, math = input().split()
    kor = int(kor)
    eng = int(eng)
    math = int(math)
    students.append(student(name, kor, eng, math))

students.sort(key=lambda x : (-x.kor, -x.eng, -x.math))

for student in students:
    print(student.name, student.kor, student.eng, student.math)

File: 240630/국영수 순이지/korean-english-math-order.js
class Student{
    constructor(name, kor, eng, math){
        this.name = name
        this.kor = kor
        this.eng = eng
        this.math = math
    }
}

const fs = require('fs')
let input = fs.readFileSync(0).toString().trim().split('\n')

let n = Number(input[0])
let students = []
for(let i =1; i<n+1; i++){
    let [n,k,e,m] = input[i].split(' ')
    students.push(new Student(n,Number(k),Number(e),Number(m)))
}


students.sort((a,b) =>{
    if(a.kor == b.kor){
        if(a.eng == b.eng){
            return b.math - a.math
        }
        return b.eng - a.eng
    }
    return b.kor - a.kor
})

for(let i = 0; i<n; i++){
    console.log(students[i].name, students[i].kor, students[i].eng, students[i].math)
}